### DISCUSSION TITLE
WalletConnect Error: No matching key. session topic doesn't exist

### DISCUSSION THREAD

**Discussion by serenae-fansubs** [rainbow-me/rainbowkit#1772] [Q&A] [answered]
I'm having trouble using Rainbowkit 2.0.0 to connect via WalletConnect from Safe App.

Using:
* Rainbowkit 2.0.0
* Wagmi 2.5.7
* Viem 2.7.8
* Next.js 14.0.4
* React 18.2.0

I see that Rainbowkit 2.0.0 is only days old. Is it perhaps currently not yet able to handle connecting via WalletConnect v2? Is anyone else running into this issue?

```
Unhandled Runtime Error
Error: No matching key. session topic doesn't exist: 464e813cba28bdd9f02ac6725a91e930778fcab70931a5e758aa13008255d188

Call Stack
ds.isValidSessionTopic
node_modules/@walletconnect/sign-client/dist/index.es.js (1:30438)
ds.isValidEmit
node_modules/@walletconnect/sign-client/dist/index.es.js (1:27204)
ds.onSessionEventRequest
node_modules/@walletconnect/sign-client/dist/index.es.js (1:21682)
ds.processRequest
node_modules/@walletconnect/sign-client/dist/index.es.js (1:15394)
ds.processRequestsQueue
node_modules/@walletconnect/sign-client/dist/index.es.js (1:14756)
ds.onRelayEventRequest
node_modules/@walletconnect/sign-client/dist/index.es.js (1:14319)
EventEmitter.eval
node_modules/@walletconnect/sign-client/dist/index.es.js (1:28885)
```

Sometimes when connecting, I also get this additional error:

```
Unhandled Runtime Error
Error: Please call connect() before request()

Call Stack
lr.request
node_modules/@walletconnect/universal-provider/dist/index.es.js (27:27672)
v.request
node_modules/@walletconnect/ethereum-provider/dist/index.es.js (1:2509)
v.switchEthereumChain
node_modules/@walletconnect/ethereum-provider/dist/index.es.js (1:4941)
v.setChainId
node_modules/@walletconnect/ethereum-provider/dist/index.es.js (1:5471)
EventEmitter.eval
node_modules/@walletconnect/ethereum-provider/dist/index.es.js (1:4078)
EventEmitter.emit
node_modules/events/events.js (153:0)
EventEmitter.eval
node_modules/@walletconnect/universal-provider/dist/index.es.js (27:32454)
EventEmitter.emit
node_modules/events/events.js (153:0)
ds.onSessionEventRequest
node_modules/@walletconnect/sign-client/dist/index.es.js (1:21710)
ds.processRequest
node_modules/@walletconnect/sign-client/dist/index.es.js (1:15394)
ds.processRequestsQueue
node_modules/@walletconnect/sign-client/dist/index.es.js (1:14756)
ds.onRelayEventRequest
node_modules/@walletconnect/sign-client/dist/index.es.js (1:14319)
EventEmitter.eval
node_modules/@walletconnect/sign-client/dist/index.es.js (1:28885)
```

I am initializing Rainbowkit in a pretty normal manner I think:

```
const config = getDefaultConfig({
  appName: '...',
  projectId: '...',
  ssr: true,
  chains: [mainnet, goerli, sepolia],
  transports: {
    [mainnet.id]: http(),
    [goerli.id]: http(),
    [sepolia.id]: http(),
  },
})

const queryClient = new QueryClient()

function App({ Component, pageProps }) {
  return (
    <WagmiProvider config={config}>
      <QueryClientProvider client={queryClient}>
        <RainbowKitProvider>
          <Component {...pageProps} />
        </RainbowKitProvider>
      </QueryClientProvider>
    </WagmiProvider>
  )
}

export default App
```

**Comment by 0xmkh** [comment]
@serenae-fansubs Are you using VPN when testing this ? Can you check your WalletConnect dashboard and make sure the allowlist domains are correctly set ?

**Reply by serenae-fansubs** [reply]
No VPN. I currently do not have any allowlist, which I (perhaps incorrectly) assumed meant that anything is allowed:

![image](https://github.com/rainbow-me/rainbowkit/assets/95138671/8e47c461-4c65-42d6-b2c9-ea479544b998)

However, right now I am also testing in a local environment, via localhost:3000, and WalletConnect will not let me add a localhost URL to the allowlist. Maybe the mere fact that I'm doing localhost testing is the reason...

Edit: Well actually huh, it says right there on the page that:

> All origins are allowed by default, and using localhost is always permitted

So it sounds like it should be working then. Hmm.

**Reply by 0xmkh** [reply]
I see. I thought you might be using some preview or something. May i ask what wallet you used when getting this issue ?

**Reply by glitch-txs** [reply]
could you share a minimal repro? @serenae-fansubs

**Reply by serenae-fansubs** [reply]
It's the (Gnosis) Safe Wallet app in a desktop browser, is that what you mean?

FWIW, I've tried multiple browsers as well, like Edge, Brave, Firefox. All have the same behavior. I'm using the Rainbowkit ConnectButton and it works great for Metamask and Coinbase Wallet.

It also works when connecting via WalletConnect in Trust Wallet (latest version) from a mobile device. But I'm not sure if that uses "WalletConnect v2" or not.

The error stacktraces are coming from `@walletconnect`, but maybe it's an issue specific to Gnosis Safe, not sure. Just to check if it's just me or not, is anyone else able to successfully connect from a Gnosis Safe Wallet via WalletConnect to an app using Rainbowkit 2.0.0?

**Reply by glitch-txs** [reply]
could you try here? https://lab.web3modal.com/library/wagmi/

**Reply by serenae-fansubs** [reply]
@glitch-txs Yes, I am able to successfully connect from the Safe App to that Web3Modal page.

I'm able to reproduce this with the out-of-the-box starter app (from the README) by running: `yarn create @rainbow-me/rainbowkit`

The only thing I change is the projectId in `pages/_app.tsx`.

I start it up with `yarn dev`, visit localhost:3000, and then attempt to connect with WalletConnect from the Gnosis Safe App:

![image](https://github.com/rainbow-me/rainbowkit/assets/95138671/776da19c-e8d3-4bce-a599-ebd7ac9b4dc3)

**Reply by glitch-txs** [reply]
> Yes, I am able to successfully connect from the Safe App to that Web3Modal page.

So it works well there?

This might be because of a re-render in the init function from EthereumProvider.
Could you try choosing the right chain (Gnosis) before connecting and then try to connect?

**Reply by serenae-fansubs** [reply]
Looking a bit closer actually at https://lab.web3modal.com/library/wagmi/, the wallet appears to connect fine, but I can see the same error in the browser console:

![image](https://github.com/rainbow-me/rainbowkit/assets/95138671/5aadc767-98e8-4dda-af97-c3e21d1df553)

The same goes for https://web3modal.com/

![image](https://github.com/rainbow-me/rainbowkit/assets/95138671/c8e09c66-71f9-4a14-9b27-9daf280a664f)

Why is Gnosis the right chain? This is a safe multisig wallet on Ethereum Mainnet, it's not on or using Gnosis Chain whatsoever. Ethereum Mainnet should still be the correct chain here, and my app will only support mainnet (and testnets).

Anyway, I can try creating a new Safe multisig on Gnosis Chain and see if that works! Might take me a bit because I guess I'll need to get some xDAI or whatever to do it... But in any case, Safe App definitely does support Ethereum Mainnet, as it's the app that Safe multisig owners use to interact with web3 dapps on Mainnet.

**Reply by serenae-fansubs** [reply]
But yeah... since I see that same error on WC's very own Web3Modal pages, then it sounds like it's not a RainbowKit issue per se. Maybe either a WC issue or a Safe App issue.

**Reply by glitch-txs** [reply]
Sorry I thought it was on Gnosis. I'll bring this up to the engineering team at WalletConnect next week and work on a solution. Feel free to open an issue in our monorepo for tracking purposes. Thanks!

**Reply by serenae-fansubs** [reply, marked as answer]
Okay, done! 

https://github.com/WalletConnect/walletconnect-monorepo/issues/4240
